### PR TITLE
Test primal InfeasiblePoint case

### PR DESCRIPTION
**Pull request by blegat**

### CODE REVIEW

File: src/Utilities/mockoptimizer.jl [hunk 1]
```diff
@@ -300,6 +300,11 @@ function mock_primal!(mock::MockOptimizer, conduals::Pair...)
     mock.hasprimal = false
     mock_dual!(mock, MOI.InfeasibilityCertificate, conduals...)
 end
+function mock_primal!(mock::MockOptimizer, primstatus::MOI.ResultStatusCode, conduals::Pair...)
```
Comment (mlubin) [inline src/Utilities/mockoptimizer.jl]: Where is this method called from?

Comment (blegat) [reply src/Utilities/mockoptimizer.jl]: From test/contconic.jl lines 26, 32 and 58 (see the changes of this commit)

Comment (mlubin) [reply src/Utilities/mockoptimizer.jl]: It's not obvious to track how the call to `mock_optimize` gets routed here. Do there really need to 4 versions of `mock_primal!` with overlapping signatures?

Comment (blegat) [reply src/Utilities/mockoptimizer.jl]: Is it better with c00a217 ?

File: src/Utilities/mockoptimizer.jl [hunk 2]
```diff
@@ -286,21 +286,39 @@ end
...
-    MOI.set!(mock, MOI.VariablePrimal(), MOI.get(mock, MOI.ListOfVariableIndices()), varprim)
-    mock_dual!(mock, dual...)
-end
-function mock_primal!(mock::MockOptimizer, varprim::Vector, dual...)
-    # Feasible primal solution
-    mock_primal!(mock, MOI.FeasiblePoint, varprim, dual...)
+    mock_varprimal!(mock, varprimdual...)
 end
+# Shortcut for feasible primal solution
+mock_primal!(mock::MockOptimizer, varprim::Vector, dual...) = mock_primal!(mock, MOI.FeasiblePoint, varprim, dual...)
```
Comment (mlubin) [inline src/Utilities/mockoptimizer.jl]: Why is there a `dual` argument to `mock_primal!`?

File: src/Utilities/mockoptimizer.jl [hunk 3]
```diff
@@ -286,21 +286,39 @@ end
...
 function mock_primal!(mock::MockOptimizer, conduals::Pair...)
     # No primal solution
     mock.hasprimal = false
+    mock_varprimal!(mock, conduals...)
+end
+
+# Sets variable primal to varprim
+function mock_varprimal!(mock::MockOptimizer, varprim::Vector, dual...)
+    MOI.set!(mock, MOI.VariablePrimal(), MOI.get(mock, MOI.ListOfVariableIndices()), varprim)
+    mock_dual!(mock, dual...)
```
Comment (mlubin) [inline src/Utilities/mockoptimizer.jl]: Why does `mock_varprimal!` call `mock_dual!` which calls `mock_condual!?

Comment (mlubin) [reply src/Utilities/mockoptimizer.jl]: The reader shouldn't need to go through 6 levels of recursion to understand all of the properties that are being set.

**Comment by blegat** [comment]
I have split primal and dual in 78e464b, I am merging but we can continue the discussion on the PR for contlinear tests.

**Comment by mlubin** [comment]
It's much better now that `mock_primal!` doesn't call `mock_dual!`.

**Pull request merged**